### DISCUSSION TITLE
Captcha integration for fully self-hosted SQLPage

### DISCUSSION THREAD

**Discussion by accforgithubtest** [sqlpage/SQLPage#700] [Ideas]
Hello,

I am wondering what would be the best approach to setting up a captcha integration for a fully self-hosted sqlpage ?

By a fully self-hosted setup I mean that every component of the stack (database, reverse-proxy, etc) are all self-hosted, which therefore eliminates most of the third party captcha services.

Has anyone tried to self host a captcha service ([like this one may be](https://github.com/mCaptcha/mCaptcha)) ? 

Keen to hear your ideas / recommendations for a fully self-hosted captcha set-up for sqlpage.

**Comment by lovasoa** [comment]
Hi! I have never used a self hosted captcha system, but I think nowadays it's especially hard to make something that works against a determined attacker with a good LLM...

Let us know how it goes with mcaptcha !

What exactly is your use case?

**Comment by accforgithubtest** [comment]
I am planning to give mCaptcha a go over the coming week and see how I go !
I do not have a specific use-case in mind, however I do have a small personal application that I want to put behind login, as well as a simple captcha service as a learning exercise.

**Comment by anayrat** [comment]
Hello, Do you if it is possible to also use an external captcha service ?
Thanks

**Reply by lovasoa** [reply]
For a commercial captcha service, you could just follow their official html+JavaScript integration instructions, and use them in SQLPage as a custom component
